core/trino-main/src/main/java/io/trino/operator/scalar/CharacterStringCasts.java

```diff
@@ -121,6 +121,24 @@ public static Slice varcharToCharSaturatedFloorCast(@LiteralParameter("y") long
         return codePointsToSliceUtf8(codePoints);
     }
 
+    @ScalarOperator(OperatorType.SATURATED_FLOOR_CAST)
+    @SqlType("varchar(y)")
+    @LiteralParameters({"x", "y"})
+    public static Slice varcharToVarcharSaturatedFloorCast(@LiteralParameter("y") long y, @SqlType("varchar(x)") Slice slice)
+    {
+        IntList codePoints = toCodePoints(slice);
+
+        // if Varchar(x) value length (including spaces) is greater than y, we can just truncate it
+        if (codePoints.size() >= y) {
+            // char(y) slice representation doesn't contain trailing spaces
+            codePoints.size(Math.min((int) y, codePoints.size()));
+            trimTrailing(codePoints, ' ');
```

Comment:
I thin it's a bug, user may write literal `'xyz    '` with explicit spaces. We should not truncate them

Reply:
`char(x)` values have trailing spaces trimmed away (_must have_)

`varchar(x)` values do not, otherwise how would you know what is the actual length of the value?